portfolio/src/main/webapp/script.js

```diff
@@ -5,18 +5,57 @@ function getRandomQuote() {
     })
 }
 
+let currPage = 0;
+let numCommentsOnPage;
+
 /** Fetch comments from server and add to DOM */
 function getComments() {
     maxNumComments = document.getElementById("num-comments").value;
-    fetch('/data?num=' + maxNumComments).then(response => response.json()).then((comments) => {
+    let prevBtn = document.getElementById("prevBtn")
+
+    handleFirstPage();
+
+    fetch('/data?num=' + maxNumComments + "&page=" + currPage).then(response => response.json()).then((comments) => {
         const commentElement = document.getElementById('video-comments-container');
         commentElement.innerHTML = '';
-        for (i = 0; i < comments.length; i++) {
+        numCommentsOnPage = comments.length;
+
+        handleLastPage();
+
+        for (i = 0; i < numCommentsOnPage; i++) {
             commentElement.appendChild(createCommentElement(comments[i]));
         }
     });
 }
 
+/** If on first page of comments, disable previous button*/
+function handleFirstPage() {
+    if (currPage === 0) {
+        prevBtn.disabled = true;
+    } else {
+        prevBtn.disabled = false;
+    }
+}
+
+/** Handle scenario if on last page of comments*/
+function handleLastPage() {
+    //Disable next button if on last page
+    if (numCommentsOnPage < 5) {
+        nextBtn.disabled = true;
+    } else {
+        nextBtn.disabled = false;
+    }
+
+    //handles case when last page has no comments
+    if (numCommentsOnPage === 0) {
```

Comment:
I just want to make sure I properly understand your intent here. Say we have 10 total comments. 
Would that  mean the intent is to have 3 total pages, the first two pages with 5 comments each and the last page with the "No comments to display" text?
Or is the intent for the "No comments to display" text to only appear when we have 0 total comments, such that in this same example if we have 10 comments it would only be 2 total pages and the next page button is disabled on the second page?

Reply:
The intent behind this was for two cases: 
1. there are no comments at all and so there would be a message saying "No comments to be displayed"
2. To handle the case in which there are a total of, say, 10 comments and you are displaying 5 comments per page. That means when you are on the second page, the next button would still be enabled because the javascript does not know that we are currently on the last page. I check to see if something is the last page by checking if the current page has fewer than the max number of comments that is able to be displayed per page. The reason javascript does not know that it's on the last page comes from the Dataservlet in which I query the data and use limit and offset to get the necessary data and then store that in an arraylist of comments and then return the comments. I considered perhaps just returning a list of all the comments, and then display the appropriate ones on the javascript side. However, that would mean lots of unnecessary Comment objects are being created and stored in the arraylist on the java side. I am not sure how to get the total number of comments while also making sure that I'm not creating and storing comment information that I won't really be using because it slows down the loading of each page

Reply:
Wonderful very nice job! 
Definitely not gating, just something to consider for the future, but could we do this some way that doesn't require doing another fetch? I totally agree that it's unnecessary to be returning all of the comments, however on that same vein is there some other data that we could pass along to the front-end?